lib/index.js

```diff
@@ -2167,6 +2167,44 @@ addMethod('doto', function (f) {
 Stream.prototype.tap = Stream.prototype.doto;
 _.tap = _.doto;
 
+/**
+ * Applies a function, which must return a (possibly empty) highland stream, to each value from the source, and re-emits the
+ * source value when the function return ends.
+ *
+ * @id flatTap
+ * @section Transforms
+ * @name Stream.flatTap(f)
+ * @param {Function} f - the function to apply
+ * @api public
+ *
+ * const httpLog = H(new Promnise(res => {
+ *  setTimeout(() => {
+ *    console.log('Log');
+ *    res('Success')
+ *  }, 3000)
+ * }))
+ *
+ * _([1, 2, 3]).flatTap(httpLog)   // Will emit [1,2,3]
+ */
+
+addMethod('flatTap', function (f) {
+    return this.flatMap(function (x) {
+        return f(x).consume(function (err, y, push, next) {
+            if (err) {
+                // next();
```

Comment:
I think you should push the `err` in this case and continue on with the stream. That's what most transforms do. The user can use `stopOnError` if they want to stop the stream early.

Reply:
Okay, this is an interesting one though, as it is a tap you could argue the error should be silenced? I think I do agree with you though.

Reply:
There's definitely a lot of uses of `tap` where you want to ignore errors. I just don't like unilaterally silencing errors in library functions. That's something that developer using the library should decide.

Also, your fix isn't entirely correct. It pushes `nil` immediately after an error. In general, Highland streams don't end after the first error, so you need to call `next` instead of `push(null, _.nil)`.